src/tasks/test-execute-proposal.ts

```diff
@@ -0,0 +1,86 @@
+import { TransactionResponse } from "@ethersproject/abstract-provider";
+import { task } from "hardhat/config";
+import { HardhatRuntimeEnvironment } from "hardhat/types";
+
+import { Args as ArgsDeployment } from "../ts/lib/common-interfaces";
+
+import { generateDeployment } from "./ts/proposal";
+import { execSafeTransaction, gnosisSafeAt } from "./ts/safe";
+
+const OUTPUT_FOLDER = "./output/test-execute-proposal";
+
+interface Args extends ArgsDeployment {
+  gnosisDao?: string;
+}
+
+const setupTestExecuteProposalTask: () => void = () => {
+  task(
+    "test-execute-proposal",
+    `This script generates a proposal (like the deployment script) but also executes all transactions onchain.`,
+  )
+    .addParam(
+      "claims",
+      "Path to the CSV file that contains the list of claims to generate.",
+    )
+    .addParam(
+      "settings",
+      "Path to the JSON file that contains the deployment settings.",
+    )
+    .addOptionalParam(
```

Comment:
Actually, we can read this form the setting. We need to have it there anyways for the initalTokenHolder target

Reply:
Updated. Note that I added a return value to `generateDeployment` so that I'm sure that the way this function parses the settings is the same as all scripts parse them.